{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 00__preprocess\n",
    "\n",
    "in this notebook, i take the output from feature counts (counts per gene in HUES64 and mESCs) and prepare it for DESeq2 analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import itertools\n",
    "import pandas as pd\n",
    "import math\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import re\n",
    "import seaborn as sns\n",
    "import sys\n",
    "\n",
    "from scipy.stats import spearmanr\n",
    "\n",
    "# import utils\n",
    "sys.path.append(\"../../../utils\")\n",
    "from plotting_utils import *\n",
    "\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'svg'\n",
    "mpl.rcParams['figure.autolayout'] = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sns.set(**PAPER_PRESET)\n",
    "fontsize = PAPER_FONTSIZE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "np.random.seed(2019)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## note: should probably re-run hESC samples w/o lifted gencode annotations..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rna_seq_dir = \"../../../data/03__rna_seq/02__for_DESeq2\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## CHANGE THESE WHEN FILES MOVE\n",
    "hESC_rep1_f = \"/n/rinn_data2/users/kaia/mpra/all_evo_mpra/2019__TSS_evolution_MPRA/misc/06__rna_seq/00__HUES64/04__featurecounts/hESC_rep1.counts.txt\"\n",
    "hESC_rep2_f = \"/n/rinn_data2/users/kaia/mpra/all_evo_mpra/2019__TSS_evolution_MPRA/misc/06__rna_seq/00__HUES64/04__featurecounts/hESC_rep2.counts.txt\"\n",
    "mESC_rep1_f = \"/n/rinn_data2/users/kaia/mpra/all_evo_mpra/2019__TSS_evolution_MPRA/misc/06__rna_seq/01__mESC/04__featurecounts/mESC_rep1.counts.txt\"\n",
    "mESC_rep2_f = \"/n/rinn_data2/users/kaia/mpra/all_evo_mpra/2019__TSS_evolution_MPRA/misc/06__rna_seq/01__mESC/04__featurecounts/mESC_rep2.counts.txt\"\n",
    "mESC_rep3_f = \"/n/rinn_data2/users/kaia/mpra/all_evo_mpra/2019__TSS_evolution_MPRA/misc/06__rna_seq/01__mESC/04__featurecounts/mESC_rep3.counts.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "orth_f = \"../../../misc/01__ensembl_orthologs/ensembl96_human_mouse_orths.txt.gz\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Geneid</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Length</th>\n",
       "      <th>../00__HUES64/03__alignments/hESC_rep1/accepted_hits.bam</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972.5_1</td>\n",
       "      <td>chr1;chr1;chr1;chr1</td>\n",
       "      <td>11869;12613;12975;13221</td>\n",
       "      <td>12227;12721;13052;14409</td>\n",
       "      <td>+;+;+;+</td>\n",
       "      <td>1735</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000227232.5_1</td>\n",
       "      <td>chr1;chr1;chr1;chr1;chr1;chr1;chr1;chr1;chr1;c...</td>\n",
       "      <td>14404;15005;15796;16607;16858;17233;17606;1791...</td>\n",
       "      <td>14501;15038;15947;16765;17055;17368;17742;1806...</td>\n",
       "      <td>-;-;-;-;-;-;-;-;-;-;-</td>\n",
       "      <td>1351</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000243485.4_2</td>\n",
       "      <td>chr1;chr1;chr1</td>\n",
       "      <td>29554;30267;30976</td>\n",
       "      <td>30039;30667;31109</td>\n",
       "      <td>+;+;+</td>\n",
       "      <td>1021</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000237613.2_1</td>\n",
       "      <td>chr1;chr1;chr1</td>\n",
       "      <td>34554;35245;35721</td>\n",
       "      <td>35174;35481;36081</td>\n",
       "      <td>-;-;-</td>\n",
       "      <td>1219</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000268020.3_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>52473</td>\n",
       "      <td>53312</td>\n",
       "      <td>+</td>\n",
       "      <td>840</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Geneid                                                Chr  \\\n",
       "0  ENSG00000223972.5_1                                chr1;chr1;chr1;chr1   \n",
       "1  ENSG00000227232.5_1  chr1;chr1;chr1;chr1;chr1;chr1;chr1;chr1;chr1;c...   \n",
       "2  ENSG00000243485.4_2                                     chr1;chr1;chr1   \n",
       "3  ENSG00000237613.2_1                                     chr1;chr1;chr1   \n",
       "4  ENSG00000268020.3_1                                               chr1   \n",
       "\n",
       "                                               Start  \\\n",
       "0                            11869;12613;12975;13221   \n",
       "1  14404;15005;15796;16607;16858;17233;17606;1791...   \n",
       "2                                  29554;30267;30976   \n",
       "3                                  34554;35245;35721   \n",
       "4                                              52473   \n",
       "\n",
       "                                                 End                 Strand  \\\n",
       "0                            12227;12721;13052;14409                +;+;+;+   \n",
       "1  14501;15038;15947;16765;17055;17368;17742;1806...  -;-;-;-;-;-;-;-;-;-;-   \n",
       "2                                  30039;30667;31109                  +;+;+   \n",
       "3                                  35174;35481;36081                  -;-;-   \n",
       "4                                              53312                      +   \n",
       "\n",
       "   Length  ../00__HUES64/03__alignments/hESC_rep1/accepted_hits.bam  \n",
       "0    1735                                                  2         \n",
       "1    1351                                                 45         \n",
       "2    1021                                                  3         \n",
       "3    1219                                                  0         \n",
       "4     840                                                  0         "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hESC_rep1 = pd.read_table(hESC_rep1_f, skiprows=1)\n",
    "hESC_rep2 = pd.read_table(hESC_rep2_f, skiprows=1)\n",
    "hESC_rep1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Geneid</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Length</th>\n",
       "      <th>../01__mESC/03__alignments/mESC_rep1/accepted_hits.bam</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSMUSG00000102693.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>+</td>\n",
       "      <td>1070</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSMUSG00000064842.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>+</td>\n",
       "      <td>110</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSMUSG00000051951.5</td>\n",
       "      <td>chr1;chr1;chr1;chr1</td>\n",
       "      <td>3205901;3213439;3421702;3670552</td>\n",
       "      <td>3207317;3216968;3421901;3671498</td>\n",
       "      <td>-;-;-;-</td>\n",
       "      <td>6094</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSMUSG00000102851.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3252757</td>\n",
       "      <td>3253236</td>\n",
       "      <td>+</td>\n",
       "      <td>480</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSMUSG00000103377.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3365731</td>\n",
       "      <td>3368549</td>\n",
       "      <td>-</td>\n",
       "      <td>2819</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Geneid                  Chr                            Start  \\\n",
       "0  ENSMUSG00000102693.1                 chr1                          3073253   \n",
       "1  ENSMUSG00000064842.1                 chr1                          3102016   \n",
       "2  ENSMUSG00000051951.5  chr1;chr1;chr1;chr1  3205901;3213439;3421702;3670552   \n",
       "3  ENSMUSG00000102851.1                 chr1                          3252757   \n",
       "4  ENSMUSG00000103377.1                 chr1                          3365731   \n",
       "\n",
       "                               End   Strand  Length  \\\n",
       "0                          3074322        +    1070   \n",
       "1                          3102125        +     110   \n",
       "2  3207317;3216968;3421901;3671498  -;-;-;-    6094   \n",
       "3                          3253236        +     480   \n",
       "4                          3368549        -    2819   \n",
       "\n",
       "   ../01__mESC/03__alignments/mESC_rep1/accepted_hits.bam  \n",
       "0                                                  0       \n",
       "1                                                  0       \n",
       "2                                                  2       \n",
       "3                                                  0       \n",
       "4                                                  0       "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mESC_rep1 = pd.read_table(mESC_rep1_f, skiprows=1)\n",
    "mESC_rep2 = pd.read_table(mESC_rep2_f, skiprows=1)\n",
    "mESC_rep3 = pd.read_table(mESC_rep3_f, skiprows=1)\n",
    "mESC_rep1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Transcript stable ID</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Mouse gene stable ID</th>\n",
       "      <th>Mouse protein or transcript stable ID</th>\n",
       "      <th>Mouse gene name</th>\n",
       "      <th>%id. target Mouse gene identical to query gene</th>\n",
       "      <th>%id. query gene identical to target Mouse gene</th>\n",
       "      <th>Mouse orthology confidence [0 low, 1 high]</th>\n",
       "      <th>Mouse homology type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>ENSMUSG00000064341</td>\n",
       "      <td>ENSMUSP00000080991</td>\n",
       "      <td>mt-Nd1</td>\n",
       "      <td>77.0440</td>\n",
       "      <td>77.0440</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>ENSMUSG00000064345</td>\n",
       "      <td>ENSMUSP00000080992</td>\n",
       "      <td>mt-Nd2</td>\n",
       "      <td>57.0605</td>\n",
       "      <td>57.3913</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>ENST00000361624</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>ENSMUSG00000064351</td>\n",
       "      <td>ENSMUSP00000080993</td>\n",
       "      <td>mt-Co1</td>\n",
       "      <td>90.8382</td>\n",
       "      <td>90.6615</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>ENST00000361739</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>ENSMUSG00000064354</td>\n",
       "      <td>ENSMUSP00000080994</td>\n",
       "      <td>mt-Co2</td>\n",
       "      <td>71.3656</td>\n",
       "      <td>71.3656</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>ENST00000361851</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>ENSMUSG00000064356</td>\n",
       "      <td>ENSMUSP00000080995</td>\n",
       "      <td>mt-Atp8</td>\n",
       "      <td>45.5882</td>\n",
       "      <td>46.2687</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene stable ID Transcript stable ID Gene name Mouse gene stable ID  \\\n",
       "0  ENSG00000198888      ENST00000361390    MT-ND1   ENSMUSG00000064341   \n",
       "1  ENSG00000198763      ENST00000361453    MT-ND2   ENSMUSG00000064345   \n",
       "2  ENSG00000198804      ENST00000361624    MT-CO1   ENSMUSG00000064351   \n",
       "3  ENSG00000198712      ENST00000361739    MT-CO2   ENSMUSG00000064354   \n",
       "4  ENSG00000228253      ENST00000361851   MT-ATP8   ENSMUSG00000064356   \n",
       "\n",
       "  Mouse protein or transcript stable ID Mouse gene name  \\\n",
       "0                    ENSMUSP00000080991          mt-Nd1   \n",
       "1                    ENSMUSP00000080992          mt-Nd2   \n",
       "2                    ENSMUSP00000080993          mt-Co1   \n",
       "3                    ENSMUSP00000080994          mt-Co2   \n",
       "4                    ENSMUSP00000080995         mt-Atp8   \n",
       "\n",
       "   %id. target Mouse gene identical to query gene  \\\n",
       "0                                         77.0440   \n",
       "1                                         57.0605   \n",
       "2                                         90.8382   \n",
       "3                                         71.3656   \n",
       "4                                         45.5882   \n",
       "\n",
       "   %id. query gene identical to target Mouse gene  \\\n",
       "0                                         77.0440   \n",
       "1                                         57.3913   \n",
       "2                                         90.6615   \n",
       "3                                         71.3656   \n",
       "4                                         46.2687   \n",
       "\n",
       "   Mouse orthology confidence [0 low, 1 high] Mouse homology type  \n",
       "0                                         1.0    ortholog_one2one  \n",
       "1                                         1.0    ortholog_one2one  \n",
       "2                                         1.0    ortholog_one2one  \n",
       "3                                         1.0    ortholog_one2one  \n",
       "4                                         0.0    ortholog_one2one  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orth = pd.read_table(orth_f)\n",
    "orth.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. join counts files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>long_gene_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21266</th>\n",
       "      <td>ENSG00000205186.2_1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000205186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41539</th>\n",
       "      <td>ENSG00000263393.1_1</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000263393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26802</th>\n",
       "      <td>ENSG00000175220.11_2</td>\n",
       "      <td>3707</td>\n",
       "      <td>2346</td>\n",
       "      <td>ENSG00000175220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6469</th>\n",
       "      <td>ENSG00000233729.1_1</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000233729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28811</th>\n",
       "      <td>ENSG00000177406.4_1</td>\n",
       "      <td>93</td>\n",
       "      <td>113</td>\n",
       "      <td>ENSG00000177406</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               long_gene_id  rep1  rep2          gene_id\n",
       "21266   ENSG00000205186.2_1     0     0  ENSG00000205186\n",
       "41539   ENSG00000263393.1_1    15     7  ENSG00000263393\n",
       "26802  ENSG00000175220.11_2  3707  2346  ENSG00000175220\n",
       "6469    ENSG00000233729.1_1     2     5  ENSG00000233729\n",
       "28811   ENSG00000177406.4_1    93   113  ENSG00000177406"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hESC = hESC_rep1[[\"Geneid\", \n",
    "                  \"../00__HUES64/03__alignments/hESC_rep1/accepted_hits.bam\"]].merge(hESC_rep2[[\"Geneid\",\n",
    "                                                                                                \"../00__HUES64/03__alignments/hESC_rep2/accepted_hits.bam\"]],\n",
    "                                                                                     on=\"Geneid\")\n",
    "hESC.columns = [\"long_gene_id\", \"rep1\", \"rep2\"]\n",
    "hESC[\"gene_id\"] = hESC[\"long_gene_id\"].str.split(\".\", expand=True)[0]\n",
    "hESC.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>long_gene_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>rep3</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>47873</th>\n",
       "      <td>ENSMUSG00000015665.8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSMUSG00000015665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15110</th>\n",
       "      <td>ENSMUSG00000054675.5</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSMUSG00000054675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4765</th>\n",
       "      <td>ENSMUSG00000049044.16</td>\n",
       "      <td>42</td>\n",
       "      <td>54</td>\n",
       "      <td>87</td>\n",
       "      <td>ENSMUSG00000049044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11958</th>\n",
       "      <td>ENSMUSG00000054958.6</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>ENSMUSG00000054958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13921</th>\n",
       "      <td>ENSMUSG00000106292.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSMUSG00000106292</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                long_gene_id  rep1  rep2  rep3             gene_id\n",
       "47873   ENSMUSG00000015665.8     0     0     0  ENSMUSG00000015665\n",
       "15110   ENSMUSG00000054675.5     8     4     1  ENSMUSG00000054675\n",
       "4765   ENSMUSG00000049044.16    42    54    87  ENSMUSG00000049044\n",
       "11958   ENSMUSG00000054958.6     2     2     2  ENSMUSG00000054958\n",
       "13921   ENSMUSG00000106292.1     0     0     0  ENSMUSG00000106292"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mESC = mESC_rep1[[\"Geneid\", \n",
    "                  \"../01__mESC/03__alignments/mESC_rep1/accepted_hits.bam\"]].merge(mESC_rep2[[\"Geneid\",\n",
    "                                                                                              \"../01__mESC/03__alignments/mESC_rep2/accepted_hits.bam\"]],\n",
    "                                                                                   on=\"Geneid\").merge(mESC_rep3[[\"Geneid\",\n",
    "                                                                                                                 \"../01__mESC/03__alignments/mESC_rep3/accepted_hits.bam\"]],\n",
    "                                                                                                      on=\"Geneid\")\n",
    "mESC.columns = [\"long_gene_id\", \"rep1\", \"rep2\", \"rep3\"]\n",
    "mESC[\"gene_id\"] = mESC[\"long_gene_id\"].str.split(\".\", expand=True)[0]\n",
    "mESC.sample(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. do some very quick QC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ortholog_one2one      132285\n",
       "ortholog_one2many      16840\n",
       "ortholog_many2many     13324\n",
       "Name: Mouse homology type, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orth[\"Mouse homology type\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Mouse gene stable ID</th>\n",
       "      <th>Mouse gene name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18291</th>\n",
       "      <td>ENSG00000181449</td>\n",
       "      <td>SOX2</td>\n",
       "      <td>ENSMUSG00000074637</td>\n",
       "      <td>Sox2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23293</th>\n",
       "      <td>ENSG00000204531</td>\n",
       "      <td>POU5F1</td>\n",
       "      <td>ENSMUSG00000024406</td>\n",
       "      <td>Pou5f1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31340</th>\n",
       "      <td>ENSG00000163508</td>\n",
       "      <td>EOMES</td>\n",
       "      <td>ENSMUSG00000032446</td>\n",
       "      <td>Eomes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107084</th>\n",
       "      <td>ENSG00000111704</td>\n",
       "      <td>NANOG</td>\n",
       "      <td>ENSMUSG00000012396</td>\n",
       "      <td>Nanog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene stable ID Gene name Mouse gene stable ID Mouse gene name\n",
       "18291   ENSG00000181449      SOX2   ENSMUSG00000074637            Sox2\n",
       "23293   ENSG00000204531    POU5F1   ENSMUSG00000024406          Pou5f1\n",
       "31340   ENSG00000163508     EOMES   ENSMUSG00000032446           Eomes\n",
       "107084  ENSG00000111704     NANOG   ENSMUSG00000012396           Nanog"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orth[orth[\"Gene name\"].isin([\"POU5F1\", \"NANOG\", \n",
    "                             \"SOX2\", \"XIST\", \"EOMES\"])][[\"Gene stable ID\", \"Gene name\", \n",
    "                                                         \"Mouse gene stable ID\", \"Mouse gene name\"]].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>long_gene_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8147</th>\n",
       "      <td>ENSG00000163508.12_1</td>\n",
       "      <td>2372</td>\n",
       "      <td>1949</td>\n",
       "      <td>ENSG00000163508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10092</th>\n",
       "      <td>ENSG00000181449.3_2</td>\n",
       "      <td>17777</td>\n",
       "      <td>13935</td>\n",
       "      <td>ENSG00000181449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15741</th>\n",
       "      <td>ENSG00000204531.16_2</td>\n",
       "      <td>54643</td>\n",
       "      <td>33314</td>\n",
       "      <td>ENSG00000204531</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29004</th>\n",
       "      <td>ENSG00000111704.10_1</td>\n",
       "      <td>6327</td>\n",
       "      <td>4139</td>\n",
       "      <td>ENSG00000111704</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               long_gene_id   rep1   rep2          gene_id\n",
       "8147   ENSG00000163508.12_1   2372   1949  ENSG00000163508\n",
       "10092   ENSG00000181449.3_2  17777  13935  ENSG00000181449\n",
       "15741  ENSG00000204531.16_2  54643  33314  ENSG00000204531\n",
       "29004  ENSG00000111704.10_1   6327   4139  ENSG00000111704"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hESC[hESC[\"gene_id\"].isin([\"ENSG00000181449\", \"ENSG00000204531\", \"ENSG00000163508\", \"ENSG00000111704\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>long_gene_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>rep3</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7612</th>\n",
       "      <td>ENSMUSG00000074637.7</td>\n",
       "      <td>23304</td>\n",
       "      <td>24661</td>\n",
       "      <td>30664</td>\n",
       "      <td>ENSMUSG00000074637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18622</th>\n",
       "      <td>ENSMUSG00000012396.12</td>\n",
       "      <td>41275</td>\n",
       "      <td>42557</td>\n",
       "      <td>53366</td>\n",
       "      <td>ENSMUSG00000012396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29266</th>\n",
       "      <td>ENSMUSG00000032446.14</td>\n",
       "      <td>496</td>\n",
       "      <td>650</td>\n",
       "      <td>814</td>\n",
       "      <td>ENSMUSG00000032446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43569</th>\n",
       "      <td>ENSMUSG00000024406.16</td>\n",
       "      <td>42454</td>\n",
       "      <td>45654</td>\n",
       "      <td>58705</td>\n",
       "      <td>ENSMUSG00000024406</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                long_gene_id   rep1   rep2   rep3             gene_id\n",
       "7612    ENSMUSG00000074637.7  23304  24661  30664  ENSMUSG00000074637\n",
       "18622  ENSMUSG00000012396.12  41275  42557  53366  ENSMUSG00000012396\n",
       "29266  ENSMUSG00000032446.14    496    650    814  ENSMUSG00000032446\n",
       "43569  ENSMUSG00000024406.16  42454  45654  58705  ENSMUSG00000024406"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mESC[mESC[\"gene_id\"].isin([\"ENSMUSG00000074637\", \"ENSMUSG00000024406\", \"ENSMUSG00000032446\", \"ENSMUSG00000012396\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "xist_human = \"ENSG00000229807\"\n",
    "sry_human = \"ENSG00000184895\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "xist_mouse = \"ENSMUSG00000086503\"\n",
    "sry_mouse = \"ENSMUSG00000069036\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>long_gene_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>48761</th>\n",
       "      <td>ENSG00000229807.10_2</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>ENSG00000229807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49843</th>\n",
       "      <td>ENSG00000184895.7_1</td>\n",
       "      <td>74</td>\n",
       "      <td>53</td>\n",
       "      <td>ENSG00000184895</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               long_gene_id  rep1  rep2          gene_id\n",
       "48761  ENSG00000229807.10_2     8     4  ENSG00000229807\n",
       "49843   ENSG00000184895.7_1    74    53  ENSG00000184895"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hESC[hESC[\"gene_id\"].isin([xist_human, sry_human])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>long_gene_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>rep3</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>47979</th>\n",
       "      <td>ENSMUSG00000086503.3</td>\n",
       "      <td>32</td>\n",
       "      <td>24</td>\n",
       "      <td>46</td>\n",
       "      <td>ENSMUSG00000086503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49034</th>\n",
       "      <td>ENSMUSG00000069036.3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSMUSG00000069036</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               long_gene_id  rep1  rep2  rep3             gene_id\n",
       "47979  ENSMUSG00000086503.3    32    24    46  ENSMUSG00000086503\n",
       "49034  ENSMUSG00000069036.3     0     0     0  ENSMUSG00000069036"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mESC[mESC[\"gene_id\"].isin([xist_mouse, sry_mouse])]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. prepare counts for DESeq2 library estimation (all genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "60211\n",
      "60211\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>45</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000237613</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000268020</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           gene_id  rep1  rep2\n",
       "0  ENSG00000223972     2     0\n",
       "1  ENSG00000227232    45    36\n",
       "2  ENSG00000243485     3     0\n",
       "3  ENSG00000237613     0     0\n",
       "4  ENSG00000268020     0     0"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hESC = hESC[[\"gene_id\", \"rep1\", \"rep2\"]].drop_duplicates(subset=\"gene_id\")\n",
    "print(len(hESC))\n",
    "print(len(hESC.gene_id.unique()))\n",
    "hESC.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "50600\n",
      "50600\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>rep3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSMUSG00000102693</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSMUSG00000064842</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSMUSG00000051951</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSMUSG00000102851</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSMUSG00000103377</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id  rep1  rep2  rep3\n",
       "0  ENSMUSG00000102693     0     0     0\n",
       "1  ENSMUSG00000064842     0     0     0\n",
       "2  ENSMUSG00000051951     2     2     2\n",
       "3  ENSMUSG00000102851     0     0     0\n",
       "4  ENSMUSG00000103377     0     0     0"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mESC = mESC[[\"gene_id\", \"rep1\", \"rep2\", \"rep3\"]].drop_duplicates()\n",
    "print(len(mESC))\n",
    "print(len(mESC.gene_id.unique()))\n",
    "mESC.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>column</th>\n",
       "      <th>condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rep1</td>\n",
       "      <td>rep1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rep2</td>\n",
       "      <td>rep2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  column condition\n",
       "0   rep1      rep1\n",
       "1   rep2      rep2"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hESC_cols = {\"rep1\": [\"rep1\"], \"rep2\": [\"rep2\"]}\n",
    "hESC_cols = pd.DataFrame.from_dict(hESC_cols, orient=\"index\").reset_index()\n",
    "hESC_cols.columns = [\"column\", \"condition\"]\n",
    "hESC_cols.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>column</th>\n",
       "      <th>condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rep1</td>\n",
       "      <td>rep1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rep2</td>\n",
       "      <td>rep2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rep3</td>\n",
       "      <td>rep3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  column condition\n",
       "0   rep1      rep1\n",
       "1   rep2      rep2\n",
       "2   rep3      rep3"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mESC_cols = {\"rep1\": [\"rep1\"], \"rep2\": [\"rep2\"], \"rep3\": [\"rep3\"]}\n",
    "mESC_cols = pd.DataFrame.from_dict(mESC_cols, orient=\"index\").reset_index()\n",
    "mESC_cols.columns = [\"column\", \"condition\"]\n",
    "mESC_cols.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. prepare counts for DESeq2 differential expression analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ortholog_one2one      132285\n",
       "ortholog_one2many      16840\n",
       "ortholog_many2many     13324\n",
       "Name: Mouse homology type, dtype: int64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orth[\"Mouse homology type\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Transcript stable ID</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Mouse gene stable ID</th>\n",
       "      <th>Mouse protein or transcript stable ID</th>\n",
       "      <th>Mouse gene name</th>\n",
       "      <th>%id. target Mouse gene identical to query gene</th>\n",
       "      <th>%id. query gene identical to target Mouse gene</th>\n",
       "      <th>Mouse orthology confidence [0 low, 1 high]</th>\n",
       "      <th>Mouse homology type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>ENSMUSG00000064341</td>\n",
       "      <td>ENSMUSP00000080991</td>\n",
       "      <td>mt-Nd1</td>\n",
       "      <td>77.0440</td>\n",
       "      <td>77.0440</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>ENSMUSG00000064345</td>\n",
       "      <td>ENSMUSP00000080992</td>\n",
       "      <td>mt-Nd2</td>\n",
       "      <td>57.0605</td>\n",
       "      <td>57.3913</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>ENST00000361624</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>ENSMUSG00000064351</td>\n",
       "      <td>ENSMUSP00000080993</td>\n",
       "      <td>mt-Co1</td>\n",
       "      <td>90.8382</td>\n",
       "      <td>90.6615</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>ENST00000361739</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>ENSMUSG00000064354</td>\n",
       "      <td>ENSMUSP00000080994</td>\n",
       "      <td>mt-Co2</td>\n",
       "      <td>71.3656</td>\n",
       "      <td>71.3656</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>ENST00000361851</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>ENSMUSG00000064356</td>\n",
       "      <td>ENSMUSP00000080995</td>\n",
       "      <td>mt-Atp8</td>\n",
       "      <td>45.5882</td>\n",
       "      <td>46.2687</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene stable ID Transcript stable ID Gene name Mouse gene stable ID  \\\n",
       "0  ENSG00000198888      ENST00000361390    MT-ND1   ENSMUSG00000064341   \n",
       "1  ENSG00000198763      ENST00000361453    MT-ND2   ENSMUSG00000064345   \n",
       "2  ENSG00000198804      ENST00000361624    MT-CO1   ENSMUSG00000064351   \n",
       "3  ENSG00000198712      ENST00000361739    MT-CO2   ENSMUSG00000064354   \n",
       "4  ENSG00000228253      ENST00000361851   MT-ATP8   ENSMUSG00000064356   \n",
       "\n",
       "  Mouse protein or transcript stable ID Mouse gene name  \\\n",
       "0                    ENSMUSP00000080991          mt-Nd1   \n",
       "1                    ENSMUSP00000080992          mt-Nd2   \n",
       "2                    ENSMUSP00000080993          mt-Co1   \n",
       "3                    ENSMUSP00000080994          mt-Co2   \n",
       "4                    ENSMUSP00000080995         mt-Atp8   \n",
       "\n",
       "   %id. target Mouse gene identical to query gene  \\\n",
       "0                                         77.0440   \n",
       "1                                         57.0605   \n",
       "2                                         90.8382   \n",
       "3                                         71.3656   \n",
       "4                                         45.5882   \n",
       "\n",
       "   %id. query gene identical to target Mouse gene  \\\n",
       "0                                         77.0440   \n",
       "1                                         57.3913   \n",
       "2                                         90.6615   \n",
       "3                                         71.3656   \n",
       "4                                         46.2687   \n",
       "\n",
       "   Mouse orthology confidence [0 low, 1 high] Mouse homology type  \n",
       "0                                         1.0    ortholog_one2one  \n",
       "1                                         1.0    ortholog_one2one  \n",
       "2                                         1.0    ortholog_one2one  \n",
       "3                                         1.0    ortholog_one2one  \n",
       "4                                         0.0    ortholog_one2one  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# remove many2many orths\n",
    "orth_sub = orth[orth[\"Mouse homology type\"] != \"ortholog_many2many\"]\n",
    "orth_sub.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id_human</th>\n",
       "      <th>gene_name_human</th>\n",
       "      <th>gene_id_mouse</th>\n",
       "      <th>gene_name_mouse</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>ENSMUSG00000064341</td>\n",
       "      <td>mt-Nd1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>ENSMUSG00000064345</td>\n",
       "      <td>mt-Nd2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>ENSMUSG00000064351</td>\n",
       "      <td>mt-Co1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>ENSMUSG00000064354</td>\n",
       "      <td>mt-Co2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>ENSMUSG00000064356</td>\n",
       "      <td>mt-Atp8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_id_human gene_name_human       gene_id_mouse gene_name_mouse\n",
       "0  ENSG00000198888          MT-ND1  ENSMUSG00000064341          mt-Nd1\n",
       "1  ENSG00000198763          MT-ND2  ENSMUSG00000064345          mt-Nd2\n",
       "2  ENSG00000198804          MT-CO1  ENSMUSG00000064351          mt-Co1\n",
       "3  ENSG00000198712          MT-CO2  ENSMUSG00000064354          mt-Co2\n",
       "4  ENSG00000228253         MT-ATP8  ENSMUSG00000064356         mt-Atp8"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# subset to genes only\n",
    "orth_genes = orth_sub[[\"Gene stable ID\", \"Gene name\", \"Mouse gene stable ID\", \"Mouse gene name\"]].drop_duplicates()\n",
    "orth_genes.columns = [\"gene_id_human\", \"gene_name_human\", \"gene_id_mouse\", \"gene_name_mouse\"]\n",
    "orth_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id_human</th>\n",
       "      <th>gene_name_human</th>\n",
       "      <th>gene_id_mouse</th>\n",
       "      <th>gene_name_mouse</th>\n",
       "      <th>orth_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>ENSMUSG00000064341</td>\n",
       "      <td>mt-Nd1</td>\n",
       "      <td>ENSG00000198888__ENSMUSG00000064341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>ENSMUSG00000064345</td>\n",
       "      <td>mt-Nd2</td>\n",
       "      <td>ENSG00000198763__ENSMUSG00000064345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>ENSMUSG00000064351</td>\n",
       "      <td>mt-Co1</td>\n",
       "      <td>ENSG00000198804__ENSMUSG00000064351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>ENSMUSG00000064354</td>\n",
       "      <td>mt-Co2</td>\n",
       "      <td>ENSG00000198712__ENSMUSG00000064354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>ENSMUSG00000064356</td>\n",
       "      <td>mt-Atp8</td>\n",
       "      <td>ENSG00000228253__ENSMUSG00000064356</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_id_human gene_name_human       gene_id_mouse gene_name_mouse  \\\n",
       "0  ENSG00000198888          MT-ND1  ENSMUSG00000064341          mt-Nd1   \n",
       "1  ENSG00000198763          MT-ND2  ENSMUSG00000064345          mt-Nd2   \n",
       "2  ENSG00000198804          MT-CO1  ENSMUSG00000064351          mt-Co1   \n",
       "3  ENSG00000198712          MT-CO2  ENSMUSG00000064354          mt-Co2   \n",
       "4  ENSG00000228253         MT-ATP8  ENSMUSG00000064356         mt-Atp8   \n",
       "\n",
       "                               orth_id  \n",
       "0  ENSG00000198888__ENSMUSG00000064341  \n",
       "1  ENSG00000198763__ENSMUSG00000064345  \n",
       "2  ENSG00000198804__ENSMUSG00000064351  \n",
       "3  ENSG00000198712__ENSMUSG00000064354  \n",
       "4  ENSG00000228253__ENSMUSG00000064356  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orth_genes[\"orth_id\"] = orth_genes[\"gene_id_human\"] + \"__\" + orth_genes[\"gene_id_mouse\"]\n",
    "orth_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>gene_id_human</th>\n",
       "      <th>gene_name_human</th>\n",
       "      <th>gene_id_mouse</th>\n",
       "      <th>gene_name_mouse</th>\n",
       "      <th>orth_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000187634</td>\n",
       "      <td>91</td>\n",
       "      <td>67</td>\n",
       "      <td>ENSG00000187634</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>ENSMUSG00000096351</td>\n",
       "      <td>Samd11</td>\n",
       "      <td>ENSG00000187634__ENSMUSG00000096351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>7273</td>\n",
       "      <td>4400</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>ENSMUSG00000095567</td>\n",
       "      <td>Noc2l</td>\n",
       "      <td>ENSG00000188976__ENSMUSG00000095567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000187961</td>\n",
       "      <td>975</td>\n",
       "      <td>697</td>\n",
       "      <td>ENSG00000187961</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>ENSMUSG00000078485</td>\n",
       "      <td>Plekhn1</td>\n",
       "      <td>ENSG00000187961__ENSMUSG00000078485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000187642</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>ENSG00000187642</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>ENSMUSG00000078486</td>\n",
       "      <td>Perm1</td>\n",
       "      <td>ENSG00000187642__ENSMUSG00000078486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000187608</td>\n",
       "      <td>546</td>\n",
       "      <td>484</td>\n",
       "      <td>ENSG00000187608</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>ENSMUSG00000035692</td>\n",
       "      <td>Isg15</td>\n",
       "      <td>ENSG00000187608__ENSMUSG00000035692</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           gene_id  rep1  rep2    gene_id_human gene_name_human  \\\n",
       "0  ENSG00000187634    91    67  ENSG00000187634          SAMD11   \n",
       "1  ENSG00000188976  7273  4400  ENSG00000188976           NOC2L   \n",
       "2  ENSG00000187961   975   697  ENSG00000187961          KLHL17   \n",
       "3  ENSG00000187642    11     4  ENSG00000187642           PERM1   \n",
       "4  ENSG00000187608   546   484  ENSG00000187608           ISG15   \n",
       "\n",
       "        gene_id_mouse gene_name_mouse                              orth_id  \n",
       "0  ENSMUSG00000096351          Samd11  ENSG00000187634__ENSMUSG00000096351  \n",
       "1  ENSMUSG00000095567           Noc2l  ENSG00000188976__ENSMUSG00000095567  \n",
       "2  ENSMUSG00000078485         Plekhn1  ENSG00000187961__ENSMUSG00000078485  \n",
       "3  ENSMUSG00000078486           Perm1  ENSG00000187642__ENSMUSG00000078486  \n",
       "4  ENSMUSG00000035692           Isg15  ENSG00000187608__ENSMUSG00000035692  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hESC_orth = hESC.merge(orth_genes, left_on=\"gene_id\", right_on=\"gene_id_human\")\n",
    "hESC_orth.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19978\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orth_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000187634__ENSMUSG00000096351</td>\n",
       "      <td>91</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000188976__ENSMUSG00000095567</td>\n",
       "      <td>7273</td>\n",
       "      <td>4400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000187961__ENSMUSG00000078485</td>\n",
       "      <td>975</td>\n",
       "      <td>697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000187642__ENSMUSG00000078486</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000187608__ENSMUSG00000035692</td>\n",
       "      <td>546</td>\n",
       "      <td>484</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               orth_id  rep1  rep2\n",
       "0  ENSG00000187634__ENSMUSG00000096351    91    67\n",
       "1  ENSG00000188976__ENSMUSG00000095567  7273  4400\n",
       "2  ENSG00000187961__ENSMUSG00000078485   975   697\n",
       "3  ENSG00000187642__ENSMUSG00000078486    11     4\n",
       "4  ENSG00000187608__ENSMUSG00000035692   546   484"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hESC_orth = hESC_orth[[\"orth_id\", \"rep1\", \"rep2\"]].drop_duplicates()\n",
    "print(len(hESC_orth))\n",
    "hESC_orth.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>rep3</th>\n",
       "      <th>gene_id_human</th>\n",
       "      <th>gene_name_human</th>\n",
       "      <th>gene_id_mouse</th>\n",
       "      <th>gene_name_mouse</th>\n",
       "      <th>orth_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSMUSG00000064842</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000252998</td>\n",
       "      <td>RNU6-889P</td>\n",
       "      <td>ENSMUSG00000064842</td>\n",
       "      <td>Gm26206</td>\n",
       "      <td>ENSG00000252998__ENSMUSG00000064842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSMUSG00000051951</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>ENSG00000206579</td>\n",
       "      <td>XKR4</td>\n",
       "      <td>ENSMUSG00000051951</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>ENSG00000206579__ENSMUSG00000051951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSMUSG00000025900</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000104237</td>\n",
       "      <td>RP1</td>\n",
       "      <td>ENSMUSG00000025900</td>\n",
       "      <td>Rp1</td>\n",
       "      <td>ENSG00000104237__ENSMUSG00000025900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSMUSG00000025902</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000164736</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>ENSMUSG00000025902</td>\n",
       "      <td>Sox17</td>\n",
       "      <td>ENSG00000164736__ENSMUSG00000025902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSMUSG00000096126</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000200887</td>\n",
       "      <td>RNU6-598P</td>\n",
       "      <td>ENSMUSG00000096126</td>\n",
       "      <td>Gm22307</td>\n",
       "      <td>ENSG00000200887__ENSMUSG00000096126</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id  rep1  rep2  rep3    gene_id_human gene_name_human  \\\n",
       "0  ENSMUSG00000064842     0     0     0  ENSG00000252998       RNU6-889P   \n",
       "1  ENSMUSG00000051951     2     2     2  ENSG00000206579            XKR4   \n",
       "2  ENSMUSG00000025900     0     0     0  ENSG00000104237             RP1   \n",
       "3  ENSMUSG00000025902    14    18    26  ENSG00000164736           SOX17   \n",
       "4  ENSMUSG00000096126     0     0     0  ENSG00000200887       RNU6-598P   \n",
       "\n",
       "        gene_id_mouse gene_name_mouse                              orth_id  \n",
       "0  ENSMUSG00000064842         Gm26206  ENSG00000252998__ENSMUSG00000064842  \n",
       "1  ENSMUSG00000051951            Xkr4  ENSG00000206579__ENSMUSG00000051951  \n",
       "2  ENSMUSG00000025900             Rp1  ENSG00000104237__ENSMUSG00000025900  \n",
       "3  ENSMUSG00000025902           Sox17  ENSG00000164736__ENSMUSG00000025902  \n",
       "4  ENSMUSG00000096126         Gm22307  ENSG00000200887__ENSMUSG00000096126  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mESC_orth = mESC.merge(orth_genes, left_on=\"gene_id\", right_on=\"gene_id_mouse\")\n",
    "mESC_orth.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20379\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orth_id</th>\n",
       "      <th>rep1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>rep3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000252998__ENSMUSG00000064842</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000206579__ENSMUSG00000051951</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000104237__ENSMUSG00000025900</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000164736__ENSMUSG00000025902</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000200887__ENSMUSG00000096126</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               orth_id  rep1  rep2  rep3\n",
       "0  ENSG00000252998__ENSMUSG00000064842     0     0     0\n",
       "1  ENSG00000206579__ENSMUSG00000051951     2     2     2\n",
       "2  ENSG00000104237__ENSMUSG00000025900     0     0     0\n",
       "3  ENSG00000164736__ENSMUSG00000025902    14    18    26\n",
       "4  ENSG00000200887__ENSMUSG00000096126     0     0     0"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mESC_orth = mESC_orth[[\"orth_id\", \"rep1\", \"rep2\", \"rep3\"]].drop_duplicates()\n",
    "print(len(mESC_orth))\n",
    "mESC_orth.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19858\n",
      "19858\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orth_id</th>\n",
       "      <th>hESC_rep1</th>\n",
       "      <th>hESC_rep2</th>\n",
       "      <th>mESC_rep1</th>\n",
       "      <th>mESC_rep2</th>\n",
       "      <th>mESC_rep3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000187634__ENSMUSG00000096351</td>\n",
       "      <td>91</td>\n",
       "      <td>67</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000188976__ENSMUSG00000095567</td>\n",
       "      <td>7273</td>\n",
       "      <td>4400</td>\n",
       "      <td>13774</td>\n",
       "      <td>14229</td>\n",
       "      <td>18741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000187961__ENSMUSG00000078485</td>\n",
       "      <td>975</td>\n",
       "      <td>697</td>\n",
       "      <td>657</td>\n",
       "      <td>751</td>\n",
       "      <td>929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000187642__ENSMUSG00000078486</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>28</td>\n",
       "      <td>22</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000187608__ENSMUSG00000035692</td>\n",
       "      <td>546</td>\n",
       "      <td>484</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               orth_id  hESC_rep1  hESC_rep2  mESC_rep1  \\\n",
       "0  ENSG00000187634__ENSMUSG00000096351         91         67          5   \n",
       "1  ENSG00000188976__ENSMUSG00000095567       7273       4400      13774   \n",
       "2  ENSG00000187961__ENSMUSG00000078485        975        697        657   \n",
       "3  ENSG00000187642__ENSMUSG00000078486         11          4         28   \n",
       "4  ENSG00000187608__ENSMUSG00000035692        546        484          6   \n",
       "\n",
       "   mESC_rep2  mESC_rep3  \n",
       "0          5         10  \n",
       "1      14229      18741  \n",
       "2        751        929  \n",
       "3         22         20  \n",
       "4          5          5  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orth_counts = hESC_orth.merge(mESC_orth, on=\"orth_id\", suffixes=(\"_x\", \"_y\"))\n",
    "print(len(orth_counts))\n",
    "print(len(orth_counts.orth_id.unique()))\n",
    "orth_counts.columns = [\"orth_id\", \"hESC_rep1\", \"hESC_rep2\", \"mESC_rep1\", \"mESC_rep2\", \"mESC_rep3\"]\n",
    "orth_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>column</th>\n",
       "      <th>condition</th>\n",
       "      <th>sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hESC_rep1</td>\n",
       "      <td>hESC</td>\n",
       "      <td>sample1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hESC_rep2</td>\n",
       "      <td>hESC</td>\n",
       "      <td>sample2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mESC_rep1</td>\n",
       "      <td>mESC</td>\n",
       "      <td>sample1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>mESC_rep2</td>\n",
       "      <td>mESC</td>\n",
       "      <td>sample2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mESC_rep3</td>\n",
       "      <td>mESC</td>\n",
       "      <td>sample3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      column condition   sample\n",
       "0  hESC_rep1      hESC  sample1\n",
       "1  hESC_rep2      hESC  sample2\n",
       "2  mESC_rep1      mESC  sample1\n",
       "3  mESC_rep2      mESC  sample2\n",
       "4  mESC_rep3      mESC  sample3"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orth_cols = {\"hESC_rep1\": [\"hESC\", \"sample1\"], \"hESC_rep2\": [\"hESC\", \"sample2\"], \n",
    "             \"mESC_rep1\": [\"mESC\", \"sample1\"], \"mESC_rep2\": [\"mESC\", \"sample2\"], \"mESC_rep3\": [\"mESC\", \"sample3\"]}\n",
    "orth_cols = pd.DataFrame.from_dict(orth_cols, orient=\"index\").reset_index()\n",
    "orth_cols.columns = [\"column\", \"condition\", \"sample\"]\n",
    "orth_cols.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. write files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hESC.to_csv(\"%s/hESC_all.counts.txt\" % rna_seq_dir, sep=\"\\t\", index=False)\n",
    "mESC.to_csv(\"%s/mESC_all.counts.txt\" % rna_seq_dir, sep=\"\\t\", index=False)\n",
    "hESC_cols.to_csv(\"%s/hESC_all.cols.txt\" % rna_seq_dir, sep=\"\\t\", index=False)\n",
    "mESC_cols.to_csv(\"%s/mESC_all.cols.txt\" % rna_seq_dir, sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "orth_counts.to_csv(\"%s/orths.counts.txt\" % rna_seq_dir, sep=\"\\t\", index=False)\n",
    "orth_cols.to_csv(\"%s/orths.cols.txt\" % rna_seq_dir, sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
